src/Demo.re

```diff
@@ -409,34 +449,52 @@ let playTurn = (players, shuffled) => {
   // Js.log("Team b tricks:")
   // Js.log(teamBTricks^)
 
-  scoreTrick(teamThatCalledTrump, teamATricks^, teamBTricks^, false, false);
-}
-
-let playGame = (players, deck) => {
-  while(teamAScore^ < 10 && teamBScore^ < 10) {
-    Js.log("Team A points: ")
-    Js.log(teamAScore^)
-    Js.log("Team B points: ")
-    Js.log(teamBScore^)
-    let (teamThatWonTrick, score) = playTurn(players, shuffleDeck(deck));
-    teamThatWonTrick == teamA ? teamAScore := teamAScore^ + score : teamBScore := teamBScore^ + score;
-  }
-
-  if(teamAScore^ >= 10) {
-    Js.log("Congratulations to team A!")
+  let (winnerNo, score) = scoreTrick(teamThatCalledTrump, teamATricks^, teamBTricks^, false, false);
+  let winner =
+    switch (winnerNo) {
+    | 0 => `TeamA
+    | 1 => `TeamB
+    };
+
+  (winner, score);
+};
+
+type team = {score: int};
+
+let rec playGame = (players, deck, teamA, teamB) => {
+  if (teamA.score < 10 && teamB.score < 10) {
+    Js.log2("Team A points: ", teamA.score);
+    Js.log2("Team B points: ", teamB.score);
+    let (winner, score) = playTurn(players, shuffleDeck(deck), teamA, teamB);
+
+    let (teamA, teamB) =
+      switch (winner) {
+      | `TeamA => ({score: teamA.score + score}, teamB)
```

Comment:
Instead of encoding the winner as an int, I use a Variant to make it clearer what is what.